pkg/kubectl/cmd/create_deployment.go

```diff
@@ -153,7 +165,18 @@ func createDeployment(f cmdutil.Factory, cmdOut, cmdErr io.Writer,
 	generatorName = fallbackGeneratorNameIfNecessary(generatorName, resourcesList, cmdErr)
 
 	imageNames := cmdutil.GetFlagStringSlice(cmd, "image")
-	generator, ok := generatorFromName(generatorName, imageNames, deploymentName)
+	replicas := cmdutil.GetFlagInt(cmd, "replicas")
+
+	command := []string{}
+	argsArray := args[1:]
```

Comment:
Good catch. I do not have a test on this. I will write one and fix any bugs I find.